src/Properties/BadStatesProperty.jl

```diff
@@ -0,0 +1,55 @@
+"""
+    BadStatesProperty{N<:Real} <: Property
+
+Type that represents a safety property characterized by a set of bad states.
+The property is satisfied by a given set of states if the intersection with the
+set of bad states is empty.
+
+### Fields
+
+- `bad`     -- convex set representing the bad states
+- `witness` -- witness point (empty vector if not set)
+
+### Notes
+
+The following formula characterizes whether a set ``X`` satisfies a safety
+property characterized by a set of bad states 𝑃:
+
+```math
+    X \\models 𝑃 \\iff X ∩ 𝑃.\\texttt{bad} = ∅
+```
+"""
+mutable struct BadStatesProperty{N<:Real} <: Property
+    bad::LazySet
+    witness::Vector{N}
+
+    BadStatesProperty{N}(bad::LazySet) where {N<:Real} = new(bad, N[])
+end
+
+# type-less convenience constructor
+BadStatesProperty(bad::LazySet{N}) where {N<:Real} =
+    BadStatesProperty{N}(bad)
+
+"""
+    check(𝑃::BadStatesProperty, X::LazySet)::Bool
+
+Checks whether a convex set is disjoint from the set of bad states.
+
+### Input
+
+- `𝑃` -- safety property with bad states
+- `X` -- convex set
+
+### Output
+
+`true` iff the given set of states does not intersect with the set of bad
+states.
+"""
+@inline function check(𝑃::BadStatesProperty, X::LazySet)::Bool
+    empty_intersection, witness = is_intersection_empty(X, 𝑃.bad, true)
```

Comment:
I was thinking that, since producing witness is in general more expensive, if it is more convenient to do:

```julia
empty_intersection = is_intersection_empty(X, 𝑃.bad, false)

if !empty_intersection
    empty_intersection, witness = is_intersection_empty(X, 𝑃.bad, true)
    𝑃.witness = witness
end
```

The downside is that `is_intersection_empty` is called twice the property is violated; the upside is that this function can be made faster. In the former case it is done only once, but if the property is satisfied, `check` is usually called many times in a tight loop.

Worth trying in a real reachability problem.

Tested independently:

```julia
julia> using LazySets; R = rand(Hyperrectangle, dim=100); H = rand(HalfSpace, dim=100);

julia> @btime is_intersection_empty($R, $H, false)
  272.878 ns (3 allocations: 1.77 KiB)
false

julia> @btime is_intersection_empty($R, $H, true);
  771.026 ns (5 allocations: 1.81 KiB)
```

```julia
julia> R = rand(Hyperrectangle, dim=2); H = rand(HalfSpace, dim=2);

julia> @btime is_intersection_empty($R, $H, false);
  152.680 ns (3 allocations: 208 bytes)

julia> @btime is_intersection_empty($R, $H, true);
  589.767 ns (6 allocations: 336 bytes)
```

Reply:
I have already created #495 for that, but I want to go in steps. Let us keep this PR clean of further experiments.

Reply:
Excellent! The above experiment gives 2~3x improvement for a single call, so i expect that this is interesting for reachability#check mode..